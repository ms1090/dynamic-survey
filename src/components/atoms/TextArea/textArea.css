.text-area {
  display: block;
}
.text-area textarea {
  border: solid 1px #8c8c8a;
  border-radius: 3px;
  box-sizing: border-box;
  color: #5d5d5d;
  display: block;
  height: 209px;
  margin: 10px 0;
  padding: 10px 16px;
  resize: none;
  width: 100%;
}