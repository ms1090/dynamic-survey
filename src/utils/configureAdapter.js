import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
// configure adapter to use enzyme in unit test cases
configure({ adapter: new Adapter() });
