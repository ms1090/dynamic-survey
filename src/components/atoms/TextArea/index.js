import React from 'react';
import PropTypes from 'prop-types';
import './textArea.css';

/**
 * Display textarea element as per customization props
 * @param {String} className CSS class name
 * @param {String} id Textarea element id
 * @param {String} label Label for textarea element
 * @return {React.Element} Return textarea element UI
 */
const TextArea = ({
  className,
  id,
  label,
  ...others
}) => (
  /* eslint-disable jsx-a11y/label-has-for */
  <label className={`text-area ${className}`} htmlFor={id}>
    <span>{label}</span>
    <textarea
      id={id}
      {...others}
    />
  </label>
);
// default values for component props
TextArea.defaultProps = {
  className: '',
  id: '',
  label: '',
};
// prop types checking
TextArea.propTypes = {
  className: PropTypes.string,
  id: PropTypes.string,
  label: PropTypes.string,
};
export default TextArea;
