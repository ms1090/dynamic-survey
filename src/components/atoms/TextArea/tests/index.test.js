import React from 'react';
import { shallow } from 'enzyme';
import TextArea from '..';
import toJson from 'enzyme-to-json';
import 'utils/configureAdapter';

describe('TextArea Component', () => {
  test('should render correctly', () => {
    const wrapper = shallow(<TextArea
      id="testId"
      className="test-className"
      label="test label"
    />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
