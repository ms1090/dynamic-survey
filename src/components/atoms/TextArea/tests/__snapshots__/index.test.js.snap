// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea Component should render correctly 1`] = `
<label
  className="text-area test-className"
  htmlFor="testId"
>
  <span>
    test label
  </span>
  <textarea
    id="testId"
  />
</label>
`;
